### PR TITLE
Add willShowViewControllerSignal

### PR DESCRIPTION
**Pull request by Iamsamwhoami**
This PR adds a signal that signals just before showing a view controller on a UINavigationController.

### CODE REVIEW

**Comment by mansbernhardt** [comment]
@Iamsamwhoami Could you add an entry to the changelog as well?

**Comment by mansbernhardt** [comment]
@nataliq I can see the test `testNavigationBarVisibilityPreference` fails for this PR. I can't see how the addition in this PR could affect any tests. I think #29 was closed for similar reasons. It would be unfortunate if these tests blocks new features coming in if they are false positives as it will discourage people to contribute. It would be great if you could investigate this further.

**Comment by nataliq** [comment]
agreed, i opened https://github.com/iZettle/Presentation/pull/37

**Comment by Iamsamwhoami** [comment]
@mansbernhardt #29 was definitely failing because of a real issue, was super hard to track down tough and I couldn't spend more time on it right now.

@nataliq thanks for disabling the test!

I've added a changelog entry now, so are we ready to get this merged?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# 1.4.1
+- Add `willShowViewControllerSignal` to `UINavigationController` that reflects UINavigationControllerDelegate's `navigationController(_:willShow:animated:)` delegate method.
```
Comment (mansbernhardt) [inline CHANGELOG.md]: Would be great if you bumped the project and pod file version as well.

Comment (Iamsamwhoami) [reply CHANGELOG.md]: @mansbernhardt Sorted!

**Comment by mansbernhardt** [comment]
@Iamsamwhoami Do you have merge privileges or should I merge this PR for you?

**Comment by Iamsamwhoami** [comment]
@mansbernhardt merged now 💃

**Pull request merged**